src/serialization/interval_log/mod.rs

```diff
@@ -0,0 +1,588 @@
+//! let count = iter.map(|r| r.unwrap())
+//!     // only look at intervals (which are the only non-comment lines in this log)
+//!     .filter_map(|e| match e {
+//!         interval_log::LogEntry::Interval(ilh) => Some(ilh),
+//!          _ => None
+//!     })
+//!     // do any filtering you want
+//!     .filter(|ilh| ilh.start_timestamp() >= 3.0)
+//!     .count();
+//!
+//! assert_eq!(3, count);
+//! ```
+//!
+//! Write a log.
+//!
+//! ```
+//! use std::str;
+//! use hdrsample;
+//! use hdrsample::serialization;
+//! use hdrsample::serialization::interval_log;
+//!
+//! let mut buf = Vec::new();
+//! let mut serializer = serialization::V2Serializer::new();
+//!
+//! let mut h = hdrsample::Histogram::<u64>::new_with_bounds(
+//!     1, u64::max_value(), 3).unwrap();
+//! h.record(12345).unwrap();
+//!
+//! // limit scope of mutable borrow of `buf`
+//! {
+//!     let mut header_writer = interval_log::IntervalLogHeaderWriter::new(
+//!         &mut buf, &mut serializer);
+//!     header_writer.write_comment("Comments are great").unwrap();
+//!     header_writer.write_start_time(123.456789).unwrap();
+//!
+//!     let mut log_writer = header_writer.into_log_writer();
+//!
+//!     log_writer.write_comment(
+//!         "You can have comments anywhere in the log").unwrap();
+//!
+//!     log_writer
+//!         .write_histogram(
+//!             &h,
+//!             1.234,
+//!             5.678,
+//!             interval_log::Tag::new("im-a-tag"),
+//!             1.0)
+//!         .unwrap();
+//! }
+//!
+//! // `buf` is now full of stuff; we check for the first line
+//! assert_eq!("#Comments are great\n", &str::from_utf8(&buf).unwrap()[0..20]);
+//! ```
+
+extern crate base64;
+
+use std::{fmt, io, ops, str};
+use std::fmt::Write;
+
+use nom::{double, line_ending, not_line_ending, IResult};
+
+use super::super::{Counter, Histogram};
+use super::Serializer;
+
+/// Write headers for an interval log.
+///
+/// This type only allows writing comments and headers. Once you're done writing those things, use
+/// `into_log_writer()` to convert this into an `IntervalLogWriter`.
+pub struct IntervalLogHeaderWriter<'a, 'b, W: 'a + io::Write, S: 'b + Serializer> {
+    internal_writer: InternalLogWriter<'a, 'b, W, S>,
+}
+
+impl<'a, 'b, W: 'a + io::Write, S: 'b + Serializer> IntervalLogHeaderWriter<'a, 'b, W, S> {
+    /// Create a new log writer that writes to `writer` and serializes histograms with `serializer`.
+    pub fn new(writer: &'a mut W, serializer: &'b mut S) -> IntervalLogHeaderWriter<'a, 'b, W, S> {
+        IntervalLogHeaderWriter {
+            internal_writer: InternalLogWriter {
+                writer,
+                serializer,
+                text_buf: String::new(),
+                serialize_buf: Vec::new(),
+            },
+        }
+    }
+
+    /// Add a comment line.
+    ///
+    /// If you do silly things like write a comment with a newline character, you'll end up with
+    /// an un-parseable log file. Don't do that.
```

Comment:
Maybe panic? Or filter silently?

Reply:
Or `.replace('\n', "\n#")`?

Reply:
I think you want [`char::is_control`](https://doc.rust-lang.org/std/primitive.char.html#method.is_control), which includes all the Unicode [control characters](http://www.fileformat.info/info/unicode/category/Cc/list.htm) (including newlines).